### PR TITLE
Add keyword arguments for Liquid shortcodes.

### LINKED ISSUE #1263: Support keyword arguments for Liquid shortcodes.
**Is your feature request related to a problem? Please describe.**

Currently, a Nunjucks shortcode or custom tag like this...
```
{% test 'arg1', 'arg2', foo='bar', baz=5 %}
```

...will pass the following arguments to its function:

```
[ 'arg1', 'arg2', { __keywords: true, foo: 'bar', baz: 5 } ]
```

This makes a lot of sense to me and I'd like to see its behavior replicated in Liquid shortcodes.

**Describe the solution you'd like**

I wrote a [parser for Liquid arguments](https://www.npmjs.com/package/liquid-args), which (mostly) matches the moo lexer currently used for Liquid shortcodes, but also supports keyword arguments and mimics the above behavior.

I can submit this as a pull request, though I think it partially depends on #1058 and how 11ty handles async argument evaluations.

**Describe alternatives you've considered**

There are some slight differences between mine and the moo parser, which I think bring it closer to Liquid syntax but which could be (theoretically?) breaking:

1. Whitespace / arg separators are `/,?[ \t\n\r]+/` instead of `/[, \t]+/`. So `arg1 ,arg2` or `arg1,,,arg2` would be invalid.
2. Numbers only allow one dot: `/[0-9]+\.?[0-9]*/` instead of `/[0-9]+\.*[0-9]*/`
3. Variables can only have periods separating valid variable names, i.e `foo.bar.baz` and not `..foobar.baz.`

I could bring it exactly in line with the moo lexer if that's preferred. I assume it might be possible to do this with moo too, though I haven't used it before.

### PR DESCRIPTION
**Pull request by dawaltconley**
Fixes #1263. This removes `moo` and switches the argument lexer/parser for Liquid shortcodes to [liquid-args](https://www.npmjs.com/package/liquid-args), to support keyword arguments with output mirroring Nunjucks kwargs.

This includes some breaking changes in how Liquid shortcodes handle the whitespace in between arguments. With `liquid-args`, whitespace can be any number of space, tab, newline, or carriage return characters (`[ \t\n\r]`), and will allow a comma at the beginning of the whitespace. That means that the following tags will parse correctly:
```
{% shortcode arg1, arg2, arg3 %}
{% shortcode arg1 arg2 arg3 %}
{% shortcode arg1,arg2,arg3 %}
{% shortcode arg1    arg2, 
    arg3 %}
```

But the following, which was previously supported, will break:
```
{% shortcode arg1,,arg2,,arg3 %}
{% shortcode arg1 , arg2 , arg3 %}
{% shortcode arg1 ,arg2 ,,arg3 %}
```

There are also some changes to how shortcodes parse numbers and dot-notation variable names. The previous `moo` parser would accept numbers like `10...78` and variable names like `..foobar.baz`. These were passed to the Liquid engine, which would throw an error for invalid Liquid. The `liquid-args` parser rejects these before passing to the engine. This shouldn't cause breaking changes but could affect error handling in some cases.

Basically the same as #1269, but updated to the latest version of 11ty.

### CODE REVIEW

**Comment by mphstudios** [comment]
@zachleat has there been any thought about merging this change or perhaps additional work to make the args lexer a configurable template engine option?

**Comment by zachleat** [comment]
Closing this one per the comment left over at #1263. https://github.com/11ty/eleventy/issues/1263#issuecomment-1340145119

Again, I’m sorry this didn’t make it!
